PubSub/src/OrderingKeyBatchJob.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\PubSub;
+
+use Google\Cloud\Core\Batch\BatchTrait;
+
+/**
+ * Represents a batch job for a single ordering key.
+ *
+ * Ordering keys must be segregated into separate RPCs. This class handles
+ * separate batch jobs, named by its ordering key. Each instance will send
+ * periodic RPCs containing one or more deferred messages. Messages are sent by
+ * the batch daemon or a shutdown function. This class defers the work of making
+ * service calls to {@see Google\Cloud\PubSub\BatchPublisher::publishDeferred}.
+ *
+ * @internal
+ */
+class OrderingKeyBatchJob
+{
+    use BatchTrait;
+
+    const ID_TEMPLATE = 'pubsub-topic-%s-ordering-key-%s';
```

Comment:
What if there is no ordering key? Could we just use `pubsub-topic-%s` in such a case?

Reply:
I had been using `default` as the trailing identifier, but after your comment thought better of it -- that's a totally valid ordering key name! Updated to use `pubsub-topic-%s` if no ordering key provided.